wallet-new/server/Cardano/Wallet/API/V1/Handlers/Accounts.hs

```diff
@@ -7,61 +7,73 @@ import           Universum
 import           Cardano.Wallet.API.V1.Types
+import           Cardano.Wallet.API.V1.Migration
 
 
 import qualified Pos.Core.Coin as Core
+import qualified Pos.Wallet.Web.Methods.Logic as V0
+import qualified Pos.Wallet.Web.Tracking as V0
 import           Servant
 import           Test.QuickCheck (arbitrary, generate, listOf1, resize)
 
-handlers :: WalletId -> Server Accounts.API
+handlers
+    :: (HasCompileInfo, HasConfigurations)
+    => WalletId -> ServerT Accounts.API MonadV1
 handlers walletId =
           deleteAccount walletId
     :<|>  getAccount walletId
     :<|>  listAccounts
     :<|>  newAccount walletId
     :<|>  updateAccount walletId
 
-deleteAccount :: WalletId -> AccountId -> Handler NoContent
-deleteAccount _ _ = return NoContent
+deleteAccount :: WalletId -> AccountId -> MonadV1 NoContent
+deleteAccount _ _ = return NoContent 
 
-getAccount :: WalletId -> AccountId -> Handler Account
-getAccount _ _ = liftIO $ generate arbitrary
+getAccount
+    :: (MonadThrow m, V0.MonadWalletLogicRead ctx m)
+    => WalletId -> AccountId -> m Account
+getAccount _ accId =
+  -- TODO (jk) We have to deal with `WalletId`
+  -- if we know how to migrate `WalletId` from `CAccount`
+    migrate accId >>= V0.fixingCachedAccModifier V0.getAccount >>= migrate
 
+--
 listAccounts :: PaginationParams
-             -> Handler (OneOf [Account] (ExtendedResponse [Account]))
+             -> MonadV1 (OneOf [Account] (ExtendedResponse [Account]))
 listAccounts PaginationParams {..} = do
   example <- liftIO $ generate (resize 3 arbitrary)
   case ppResponseFormat of
-    Extended -> return $ OneOf $ Right $
-      ExtendedResponse {
-        extData = example
-      , extMeta = Metadata {
-          metaTotalPages = 1
-        , metaPage = 1
-        , metaPerPage = 20
-        , metaTotalEntries = 3
-      }
-      }
+    Extended -> return $ OneOf $ Right
+        ExtendedResponse {
+            extData = example
+          , extMeta = Metadata {
+                  metaTotalPages = 1
+                , metaPage = 1
+                , metaPerPage = 20
+                , metaTotalEntries = 3
+              }
+          }
     _ -> return $ OneOf $ Left example
 
 -- | This is an example of how POST requests might look like.
 -- It also shows an example of how an error might look like.
 -- NOTE: This will probably change drastically as soon as we start using our
 -- custom monad as a base of the Handler stack, so the example here is just to
 -- give the idea of how it will look like on Swagger.
-newAccount :: WalletId -> Maybe Text -> AccountUpdate -> Handler Account
-newAccount w@(WalletId wId) _ AccountUpdate{..} = do
-    when (wId /= "testwallet") $ throwError $ Errors.toError Errors.WalletNotFound
+newAccount :: WalletId -> Maybe Text -> AccountUpdate -> MonadV1 Account
+newAccount (WalletId wId) _ AccountUpdate{..} = do
+    when (wId /= "testwallet") $ throwM $ Errors.toError Errors.WalletNotFound
```

Comment:
As this is now not a prototype anymore, we should remove this line, was added just to demonstrate how error handling would look like 😉